src/components/shop-cart/shop-cart.component.spec.ts

```diff
@@ -1,21 +1,123 @@
 import { ComponentFixture, TestBed } from '@angular/core/testing';
 import { ShopCartComponent } from './shop-cart.component';
+import { ProductService } from '../../services/products/product.service';
+import { PurchaseService } from '../../services/purchases/purchase.service';
+import { SaleService } from '../../services/sales/sale.service';
+import { SharedService } from '../../services/globalAttributes/shared.service';
+import { of } from 'rxjs';
 
 describe('ShopCartComponent', () => {
   let component: ShopCartComponent;
   let fixture: ComponentFixture<ShopCartComponent>;
 
   beforeEach(async () => {
     await TestBed.configureTestingModule({
-      imports: [ShopCartComponent]
+      declarations: [ShopCartComponent],
+      providers: [
+        { provide: ProductService, useValue: jasmine.createSpyObj('ProductService', ['getList', 'getById', 'update']) },
+        { provide: PurchaseService, useValue: jasmine.createSpyObj('PurchaseService', ['getList']) },
+        { provide: SaleService, useValue: jasmine.createSpyObj('SaleService', ['getList']) },
+        { provide: SharedService, useValue: jasmine.createSpyObj('SharedService', ['getGlobalVariable']) },
+      ],
     }).compileComponents();
+  });
 
+  beforeEach(() => {
     fixture = TestBed.createComponent(ShopCartComponent);
     component = fixture.componentInstance;
+    spyOn(component, 'getTotalPrice').and.callThrough();
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
-});
+
+  it('should initialize component and call getTotalPrice', () => {
```

Comment:
As it is a unit test, I would prefer to test the initialize component as a unit and create another unit test to get the total price